playbooks/common/openshift-cluster/upgrades/v3_5/validator.yml

```diff
@@ -16,3 +16,49 @@
   tasks:
   - name: Check for invalid namespaces and SDN errors
     oc_objectvalidator:
+
+  # What's all this PetSet business about?
+  #
+  # 'PetSets' were ALPHA resources in Kube <= 3.4. In >= 3.5 they are
+  # no longer supported. The BETA resource 'StatefulSets' replaces
+  # them. We can't migrate clients PetSets to
+  # StatefulSets. Additionally, Red Hat has never officially supported
+  # these resource types. Sorry users, but if you were using
+  # unsupported resources from the Kube documentation then we can't
+  # help you at this time.
+  #
+  # Reference: https://bugzilla.redhat.com/show_bug.cgi?id=1428229
+  - name: Check if legacy PetSets exist
+    oc_obj:
+      state: list
+      all_namespaces: true
+      kind: petsets
+    register: l_do_petsets_exist
+
+  - name: FAIL ON Resource migration 'PetSets' unsupported
+    fail:
+      msg: >
+
+        PetSets are ALPHA status Kube resources and are not supported
+        by Red Hat. Automatically migrating to their BETA replacement,
+        StatefulSets, in 3.5 is not supported. Please see the
+        Kubernetes "Upgrading from PetSets to StatefulSets"
+        documentation for additional information:
+        https://kubernetes.io/docs/tasks/manage-stateful-set/upgrade-pet-set-to-stateful-set/
+
+        PetSets must be removed prior to an upgrade. If you wish to
+        simply remove all PetSets, run this command as an admin: oc
+        get petsets --all-namespaces -o yaml | oc delete -f -
+        --cascale=false
+
+        Red Hat currently offers NO SUPPORT for PetSets and
+        StaticSets. We strongly encourage you to read the referenced
```

Comment:
@tbielawa This "StaticSets" is just a typo of StatefulSets, yea? 

Here's a revision of the whole warning message For Your Consideration™:

https://gist.github.com/adellape/f17d72cc8901b06fb3644096f2d50a81